src/Builder/Encoder/OperatorEncoder.php

```diff
@@ -68,60 +59,29 @@ private function encodeAsArray(OperatorInterface $value): stdClass
         return $this->wrap($value, $result);
     }
 
-    private function encodeAsDollarObject(OperatorInterface $value): stdClass
-    {
-        $result = new stdClass();
-        foreach (get_object_vars($value) as $key => $val) {
-            // Skip optional arguments. If they have a default value, it is resolved by the server.
-            if ($val === Optional::Undefined) {
-                continue;
-            }
-
-            $val = $this->recursiveEncode($val);
-
-            if ($key === 'geometry') {
-                if (is_object($val) && property_exists($val, '$geometry')) {
-                    $result->{'$geometry'} = $val->{'$geometry'};
-                } elseif (is_array($val) && array_key_exists('$geometry', $val)) {
-                    $result->{'$geometry'} = $val['$geometry'];
-                } else {
-                    $result->{'$geometry'} = $val;
-                }
-            } else {
-                $result->{'$' . $key} = $val;
-            }
-        }
-
-        return $this->wrap($value, $result);
-    }
-
     /**
-     * $group stage have a specific encoding because the _id argument is required and others are variadic
+     * Encode the value as an object with properties. Property names are
+     * mapped by the PROPERTIES constant.
      */
-    private function encodeAsGroup(OperatorInterface $value): stdClass
-    {
-        assert($value instanceof GroupStage);
-
-        $result = new stdClass();
-        $result->_id = $this->recursiveEncode($value->_id);
-
-        foreach (get_object_vars($value->field) as $key => $val) {
-            $result->{$key} = $this->recursiveEncode($val);
-        }
-
-        return $this->wrap($value, $result);
-    }
-
     private function encodeAsObject(OperatorInterface $value): stdClass
     {
         $result = new stdClass();
-        foreach (get_object_vars($value) as $key => $val) {
+        foreach ($value::PROPERTIES as $prop => $name) {
+            $val = $value->$prop;
+
             // Skip optional arguments. If they have a default value, it is resolved by the server.
             if ($val === Optional::Undefined) {
                 continue;
             }
 
-            $result->{$key} = $this->recursiveEncode($val);
+            if ($name === null) {
```

Comment:
Assuming this is the place where `mergeObject: true` is handled, a comment here would be helpful.

Reply:
👍🏻 Comment added.